### PR TITLE
add helpers preparing vm snapshots in basic states

### PR DESCRIPTION
**Pull request by jmartin-r7**
prepVMs.py will `enable_uac` and `disable_smb1` these are
some basic prep options hopefully useful for metasploit payloads.

### CODE REVIEW

File: helpers/prepVMs.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    prefix = args.keyword
+
+    if args.action.lower() not in validActions:
+        print('INVALID ACTION')
+    if args.action.lower() == 'enable_uac':
+        command = UAC_ENABLE_COMMAND
+    elif args.action.lower() == 'disable_smb1':
+        command = UAC_ENABLE_COMMAND
+
```
Comment (bwatters-r7) [inline helpers/prepVMs.py]: I think there was some copy/pasta here?

File: helpers/prepVMs.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    pool = None
+    try:
+        pool = multiprocessing.Pool(3)
+
+        signal.signal(signal.SIGINT, original_sigint_handler)
+
+        results = []
+        for vm_name in vm_list:
+            pool.apply_async(enable_uac, [args.hypervisorConfig, vm_name, command], callback=results.append)
+
```
Comment (bwatters-r7) [inline helpers/prepVMs.py]: If we are going to always call `enable_uac`, even if we are disabling smbv1, can we call it something generic like 'executeCommand' or 'fireTheLaser'?

File: helpers/prepVMs.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    for vm in vm_server.vmList:
+        if prefix in vm.vmName and WINDOWS_REQUIRED in vm.vmName:
+            vm_list.append(vm.vmName)
+
+    original_sigint_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)
+
+    pool = None
+    try:
+        pool = multiprocessing.Pool(3)
+
```
Comment (bwatters-r7) [inline helpers/prepVMs.py]: Maybe default to 3, but allow users to specify for large-scale systems?

File: helpers/prepVMs.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            vm_ready = False
+            while vm_ready is False:
+                vm_ready = vm_server.waitForVmsToBoot([vm])
+            vm.setUsername('vagrant')
+            vm.setPassword('vagrant')
+            vm.runCmdOnGuest(command)
+            time.sleep(schedule_delay)
+            vm.vmObject.ShutdownGuest()
+            time.sleep(10)
+            vm.powerOff()
```
Comment (bwatters-r7) [inline helpers/prepVMs.py]: Do we need to poweroff the VM?  Might be nice to return it back to whatever state it was in when we started?

File: helpers/prepVMs.py (hunk 3 above)
Comment (jmartin-r7) [reply helpers/prepVMs.py]: Noted, may not expand yet but like idea.

File: helpers/prepVMs.py (hunk 1 above)
Comment (jmartin-r7) [reply helpers/prepVMs.py]: yep thanks for catching

File: helpers/prepVMs.py (hunk 2 above)
Comment (jmartin-r7) [reply helpers/prepVMs.py]: thanks, forgot to rename

File: helpers/prepVMs.py (hunk 4 above)
Comment (jmartin-r7) [reply helpers/prepVMs.py]: VM has to reboot to apply the change, I since this is a helper for initial prep fine to shutdown for IMO.

Comment (jmartin-r7) [reply helpers/prepVMs.py]: Note: I intentionally did not take snapshots here and just reuse your manageSnapshots.py sample for my use case.

**Comment by bwatters-r7** [comment]
Using the baseline builder Windows 10x64 1511, this does not appear to turn off smbv1?  I added a debug print to verify that it was listing the right VM.

command:
```
python prepVMs.py -k Win10x64_1511 -a disable_smb1 'config.json' 
Win10x64_1511
APT_Win10x64_1511
3it [01:00, 36.53s/it]                                                                                                                                                   
Processing complete 2 systems updated
```


Verification on reboot:
image

**Comment by jmartin-r7** [comment]
Retested local and Windows 10 passed, however Server 2008 R2(sp0 & sp1) as well as Windows 7 (sp0) are failing to shutdown when processing 26 hosts.  Results are inconsistent when run only against two Win2008 hosts.  Action requested is found to be complete shutdown just did not execute.

VMWare task shows
```
Task
Shutdown Guest
Key
haTask-183-vim.VirtualMachine.shutdownGuest-8435

Description
Issues a command to the guest operating system to perform a clean shutdown of all services

Virtual machine:
Win2008r2x64sp1
State
Failed - Failed to power off the virtual machine: Cannot execute scripts.

Errors
Failed to power off the virtual machine: Cannot execute scripts.
```
 A mitigation for this may need to be added here.

**Comment by jmartin-r7** [comment]
Issue looks to have been status of VMware Tools on these VMs.  No changes here at this point, however it may be a good idea to build in a timeout as when this occurred the helper hung indefinitely.

**Pull request merged**